jetty-util/src/test/java/org/eclipse/jetty/util/HostPortTest.java

```diff
@@ -112,4 +151,12 @@ public void testInvalidAuthority(String authority)
             new HostPort(authority);
         });
     }
+
+    @ParameterizedTest
+    @MethodSource("invalidAuthorityProvider")
+    public void testUnsafeAuthority(String authority)
+    {
+        HostPort hostPort = HostPort.unsafe(authority);
+        assertNotNull(hostPort);
```

Comment:
We need to test more here.  We need to test if the port was correctly extracted or not.  So I think we need to pass in an expected port

Reply:
Many will be port 0, this is a test of unsafe against totally broken authorities that the normal (safe) flows will reject and not allow.

We have a separate test for valid authorities that use `unsafe(String)` which does test for valid splits (host + port)

Reply:
In fact, in `unsafe(String authority)` we don't even test to make sure that the port is valid (within range 1 thru 65535).

It could be parsed as -80, or 333444555 as its unsafe.

Reply:
Basically everything about using `unsafe(String authority)` is unsafe, it can easily allow conditions later that will never work (you cannot create a URL or URI from them, or a valid `Location` header, etc.

Reply:
I'm ok with it being totally unsafe, but we need to consistently and predictably in our unsafe handling. Currently if it looks like an IPv6 address we don't check for a : before posting the port. If it doesn't look like IPv6 then we do check for the colon. I think we should check in both cases. No colon then no port.

Reply:
Reworked the test cases and the implementation in HostPort.